design-doc.md

```diff
@@ -0,0 +1,179 @@
+
+and platforms, increase developer productivity and feature delivery.
+
+
+### When does a component go into ui-kit
+
+Ui-kit follows the rules of three, which means once a component is used in 3 different places it should be added to ui-kit.
+
+
+### Flexibility
+
+We are trying to avoid unnecessary flexibility of a component. There is always room for building on the existing component or creating a new component for a different use case. Always when adding flexibility for another use case consider adding a new component instead or ensure that the complexity of the component is not exploding.
+
+
+### Considerations
+
+
+#### Avoid weak elements
+
+Weak elements are elements which can’t nest block elements, for example, a `p` tag is not able to wrap around a div which will reduce the comparability of a component. This is why we try to rather use a strong element like `section` or `div` instead of `p`.
+
+
+#### Declare Prop Types
+
+Since the `ui-kit` is using typescript please make sure to define proper prop types for your components. This will improve the documentation, which creates dynamically and will provide better error messages for the users. And also this will provide auto-completion in some IDE’s. Describe the props as well as possible, so also define object shapes or array shapes.
+
+
+#### Never hard code HTML ID’s
+
+This might conflict at some point. If you need an id on the element expose a prop for that so the user can define it based on the application	.
+
+
+#### Use Logical Defaults
+
+Use prop defaults which make sense this will reduce the efforts needed to adopt this component.
+
+
+#### A11y Support
+
+A good start on that topic is to read [this website](https://a11yproject.com/) While creating your component think about users: - Which uses keyboards - expose tab indexes as props - Use semantic correct HTML - ARIA
+
+
+#### Configuration Objects
+
+Please consider using configuration objects as they will provide the user a lot of value by
+
+
+
+*   consistency
+*   less typing needed
+*   less error prone
+
+But please avoid using configuration Objects if you have 3 or fewer properties.
+
+A example pattern for this can be found [here](#configuration-objects).
+
+
+#### Single Responsibility Principle
+
+Keep the components simple by applying them to a single responsibility. In that way, the component will stay simple in most cases. Try not to introduce unnecessary props. Each prop is a configuration which will automatically increase the complexity. Before adding a property to an existing component ask yourself if this could be a new component.
+
+
+#### Keep Components Simple
+
+It's important to keep components as simple as possible, add logic to a component if it improves performance or quality of that component. For example, paginating a list if the number of items is exceeding a certain amount or having a show toggle to show more content like a dropout.
```

Comment:
"It's important to keep components as simple as possible, **only** add logic to a component if it improves **the** performance or **the** quality of that component."

I don't understand what the second sentence is trying to say.